fasthttpproxy/proxy_env.go

```diff
@@ -0,0 +1,108 @@
+package fasthttpproxy
+
+import (
+	"bufio"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"net/url"
+	"sync"
+	"time"
+
+	"golang.org/x/net/http/httpproxy"
+
+	"github.com/valyala/fasthttp"
+)
+
+// FasthttpProxyHTTPDialer returns a fasthttp.DialFunc that dials using
+// the the env(HTTP_PROXY, HTTPS_PROXY and NO_PROXY) configured HTTP proxy.
+//
+// Example usage:
+//	c := &fasthttp.Client{
+//		Dial: FasthttpProxyHTTPDialer(),
+//	}
+func FasthttpProxyHTTPDialer() fasthttp.DialFunc {
```

Comment:
hey! first of all, thanks for your contribution and welcome!

I think we can drop `FasthttpProxy` prefix from function names, as we already have that in the package name. Also, as we are trying to keep fasthttp as minimalistic as possible, this package looks like a good fit for an addon package. For example, we can create `github.com/fasthttp/fasthttpproxy` or `github.com/fasthttp/httpproxy`, and put it there. This would allow us to keep a smaller api footprint. What you think?

Reply:
Hi @kirillDanshin !

Added `FasthttpProxy` prefix to be consistent with current naming in the same package.

I'm ok to move the proxy to the addon. Could you pls create `github.com/fasthttp/fasthttpproxy` and i will open a merge request there? 

I would choose the `github.com/fasthttp/fasthttpproxy` to distinguis between fasthttp proxy and x/net/httpproxy packets, WDYT?

Reply:
@kirillDanshin Could you pls create the repo for this addon? i will move MR there.

Reply:
We already have the other proxy code in this repo with similar naming. I don't see why we shouldn't add this one either.
And since it's in a different package it doesn't increase the API footprint of the fasthttp itself.
Moving this code and the other proxy code to `github.com/fasthttp/fasthttpproxy` sounds nice but I feel like it's a v2 thing when we can introduce breaking changes.